datacube/storage/_load.py

```diff
@@ -26,13 +27,23 @@
 ProgressFunction = Callable[[int, int], Any]  # pylint: disable=invalid-name
 
 
-def _default_fuser(dst: np.ndarray, src: np.ndarray, dst_nodata: float) -> None:
+def _default_fuser(dst: np.ndarray, src: np.ndarray, dst_nodata) -> None:
     """ Overwrite only those pixels in `dst` with `src` that are "not valid"
 
         For every pixel in dst that equals to dst_nodata replace it with pixel
         from src.
     """
-    where_nodata = (dst == dst_nodata) if not np.isnan(dst_nodata) else np.isnan(dst)
+    if dtype_is_float(dst.dtype):
+        if dst_nodata is None or np.isnan(dst_nodata):
+            where_nodata = np.isnan(dst)
+        else:
+            where_nodata = np.isnan(dst) | (dst == dst_nodata)
+    else:
+        if dst_nodata is None:
+            where_nodata = np.full_like(dst, False, dtype=np.bool)
```

Comment:
this case is also new, and it doesn't have a test.

note that this won't trigger when testing via high-level function, since `dst_nodata` is always set in there. This function became more complex so needs a direct unittest to check all possible scenarios.
 
In fact it doesn't make sense to have `dst_nodata=None`, it's a limitation of the fuser interface see #691.